src/psyclone/psyir/nodes/omp_clauses.py

```diff
@@ -115,3 +117,223 @@ class OMPNogroupClause(Clause):
+        if self._clause_type == OMPDefaultClause.DefaultClauseTypes.SHARED:
+            clause_string = clause_string + "shared)"
+        elif self._clause_type == OMPDefaultClause.DefaultClauseTypes.NONE:
+            clause_string = clause_string + "none)"
+        elif (self._clause_type == 
+                OMPDefaultClause.DefaultClauseTypes.FIRSTPRIVATE):
+            clause_string = clause_string + "firstprivate)"
+        return clause_string
+
+
+    def __init__(self, clause_type=DefaultClauseTypes.SHARED):
+        if not isinstance(clause_type, OMPDefaultClause.DefaultClauseTypes):
+            raise TypeError(
+                    "OMPDefaultClause expected 'clause_type' argument of type "
+                    "OMPDefaultClause.DefaultClauseTypes but found '{0}'"
+                    .format(type(clause_type).__name__))
+        self._clause_type = clause_type
+        super(OMPDefaultClause, self).__init__()
+
+class OMPScheduleClause(Clause):
+    '''
+    OpenMP Schedule clause used for OMP Do Directives.
+    '''
+    _children_valid_format = "None"
+    _text_name = "OMPScheduleClause"
+    _schedule = ""
+
+    @property
+    def _clause_string(self):
+        return f"schedule({self._schedule})"
+
+    def set_schedule(self, schedule):
+        self._schedule = schedule
+
+    def __init__(self, schedule="static"):
+        self._schedule = schedule
+        super(OMPScheduleClause, self).__init__()
+
+class OMPDependClause(OperandClause):
+    '''
+    OpenMP Depend clause used for OpenMP Task directives.
+    '''
+    _children_valid_format = "[Reference]*"
+    _text_name = "OMPDependClause"
+    _clause_string = "depend"
+
+    class DependClauseTypes(Enum):
+        '''Enumeration of the different types of OMPDependClause supported
+        in PSyclone'''
+        IN=0
+        OUT=1
+        INOUT=2
+
+    def __init__(self, depend_type=DependClauseTypes.INOUT):
+        if not isinstance(depend_type, OMPDependClause.DependClauseTypes):
+            raise TypeError(
+                    "OMPDependClause expected 'depend_type' argument of type "
+                    "OMPDependClause.DependClauseTypes but found '{0}'"
+                    .format(type(depend_type).__name__))
+        self._operand = depend_type
+        super(OMPDependClause, self).__init__()
+
+    @staticmethod
+    def _validate_child(position, child):
+        '''
+         Decides whether a given child and position are valid for this node.
+         One child allowed, of type Literal.
+
+        :param int position: the position to be validated.
+        :param child: a child to be validated.
+        :type child: :py:class:`psyclone.psyir.nodes.Node`
+
+        :return: whether the given child and position are valid for this node.
+        :rtype: bool
+
+        '''
+        return isinstance(child, Reference)
+
+    @property
+    def operand(self):
+        if self._operand == OMPDependClause.DependClauseTypes.IN:
```

Comment:
Again, I think you can get the name of a member of an Enum and thus do away with this if block.

Reply:
I switched this to use name.lower()